be/src/exec/hdfs_scanner.h

```diff
@@ -324,9 +339,8 @@ class HdfsScanner {
     virtual Status do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) = 0;
     virtual void do_update_counter(HdfsScanProfile* profile);
     virtual bool is_jni_scanner() { return false; }
-    virtual void get_split_tasks(std::vector<pipeline::ScanSplitContextPtr>* split_tasks) {
-        split_tasks->swap(_split_tasks);
-    }
+    void move_split_tasks(std::vector<pipeline::ScanSplitContextPtr>* split_tasks);
+    bool has_split_tasks() const { return _scanner_ctx.has_split_tasks; }
```

Comment:
This function seems very error prone because it requires `move_split_tasks` to be called first but without any clear reminder

Reply:
There is no need to clear:
- by default, there is no split tasks, so `has_split_tasks` is false
- and if there are split tasks, this variable will be set `has_split_task` to true.

no need to set it back.